nodejs/eks/nodegroup.ts

```diff
@@ -391,59 +414,22 @@ ${customUserData}
 `;
         });
 
+    const version = pulumi.output(args.version || core.cluster.version);
+
+    // https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html
     let amiId: pulumi.Input<string> = args.amiId!;
-    let ignoreChanges: string[] = [];
-    if (args.amiId === undefined) {
-        const version = pulumi.output(args.version || core.cluster.version);
+    let amiType = "";
+    if (!args.amiId) {
+        amiType = "amazon-linux-2";
+    }
+    if (args.gpu) {
+        amiType = "amazon-linux-2-gpu";
+    }
+    if (amiType) {
```

Comment:
I'd wrap this entire block in a `if (!amiID) {` condition, and then simplify the following:
```suggestion
    const amiType = args.gpu ? "amazon-linux-2-gpu" : "amazon-linux-2";
```

Reply:
Though the ternary approach is cleaner, this assumes we will always use the SSM store, which is not the case when the user supplies a specific `amiID` to use.

Specifying the `amiID` is also mutually exclusive with `gpu` latest recommended AMI.

Reply:
Oops, missed the first part of the comment 👍